tests/integration/fuzzing_strategies/map_and_filter_test.cairo

```diff
@@ -0,0 +1,59 @@
+%lang starknet
+
+
+from starkware.cairo.common.math import assert_in_range
+
+
+@external
+func __setup__():
+    %{ max_examples(3) %}
+    return ()
+end
+
+@external
+func setup_chaining():
+    %{
+        given(
+               a = strategy.felts().filter(lambda x:  17 < x < 100).map(lambda x: -x).filter(lambda x: x < -20) 
+           )
+    %}
+    return ()
+end
+
+@external
+func test_chaining(a):
+    assert a = 3
+    return ()
+end
+
+@external
+func setup_filtering():
+    %{
+        given(
+               a = strategy.one_of( strategy.integers().filter(lambda x: 100 <= x <= 1000) ) 
```

Comment:
Can i filter `one_of`? I should as it is a strategy descriptor.

Reply:
If yes, please provide a test case. If not please adjust types

Reply:
Yes, please add a test :)